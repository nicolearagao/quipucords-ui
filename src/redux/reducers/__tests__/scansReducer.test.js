import { scansReducer } from '../scansReducer';
import { reportsTypes, scansTypes as types } from '../../constants';
import { reduxHelpers } from '../../common';

describe('ScansReducer', () => {
  it('should return the initial state', () => {
    expect(scansReducer.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [
      types.UPDATE_SCANS,
      types.MERGE_SCAN_DIALOG_SHOW,
      types.MERGE_SCAN_DIALOG_HIDE,
      types.SELECT_SCAN,
      types.DESELECT_SCAN,
      types.EXPANDED_SCAN,
      types.NOT_EXPANDED_SCAN,
      types.RESET_ACTIONS
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value
      };

      const resultState = scansReducer(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot(`defined type ${value}`);
    });
  });

  it('should handle all defined error types', () => {
    const specificTypes = [
      reportsTypes.GET_REPORTS_DOWNLOAD,
      types.GET_SCAN_CONNECTION_RESULTS,
      types.GET_SCAN_INSPECTION_RESULTS,
      types.GET_SCAN_JOB,
      types.GET_SCAN_JOBS,
      types.CANCEL_SCAN,
      types.PAUSE_SCAN,
      types.RESTART_SCAN,
      types.START_SCAN,
      types.GET_SCANS,
      types.DELETE_SCAN
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: reduxHelpers.REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            status: 0,
            statusText: 'ERROR TEST',
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = scansReducer(undefined, dispatched);

      expect({ type: reduxHelpers.REJECTED_ACTION(value), result: resultState }).toMatchSnapshot(
        `rejected types ${value}`
      );
    });
  });

  it('should handle all defined pending types', () => {
    const specificTypes = [
      reportsTypes.GET_REPORTS_DOWNLOAD,
      types.GET_SCAN_CONNECTION_RESULTS,
      types.GET_SCAN_INSPECTION_RESULTS,
      types.GET_SCAN_JOB,
      types.GET_SCAN_JOBS,
      types.CANCEL_SCAN,
      types.PAUSE_SCAN,
      types.RESTART_SCAN,
      types.START_SCAN,
      types.GET_SCANS,
      types.DELETE_SCAN
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: reduxHelpers.PENDING_ACTION(value)
      };

      const resultState = scansReducer(undefined, dispatched);

      expect({ type: reduxHelpers.PENDING_ACTION(value), result: resultState }).toMatchSnapshot(
        `pending types ${value}`
      );
    });
  });

  it('should handle all defined fulfilled types', () => {
    const specificTypes = [
      reportsTypes.GET_REPORTS_DOWNLOAD,
      types.GET_SCAN_CONNECTION_RESULTS,
      types.GET_SCAN_INSPECTION_RESULTS,
      types.GET_SCAN_JOB,
      types.GET_SCAN_JOBS,
      types.CANCEL_SCAN,
      types.PAUSE_SCAN,
      types.RESTART_SCAN,
      types.START_SCAN,
      types.GET_SCANS,
      types.DELETE_SCAN
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: reduxHelpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            test: 'success'
          }
        }
      };

      const resultState = scansReducer(undefined, dispatched);

      expect({ type: reduxHelpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot(
        `fulfilled types ${value}`
      );
    });
  });
});
