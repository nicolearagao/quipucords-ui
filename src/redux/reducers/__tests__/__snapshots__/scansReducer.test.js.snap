// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScansReducer should handle all defined error types: rejected types CANCEL_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "CANCEL_SCAN_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types DELETE_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "DELETE_SCAN_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types GET_REPORTS_DOWNLOAD 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_REPORTS_DOWNLOAD_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types GET_SCAN_CONNECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_CONNECTION_RESULTS_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types GET_SCAN_INSPECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_INSPECTION_RESULTS_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types GET_SCAN_JOB 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_JOB_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types GET_SCAN_JOBS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_JOBS_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types GET_SCANS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
  },
  "type": "GET_SCANS_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types PAUSE_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "PAUSE_SCAN_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types RESTART_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "RESTART_SCAN_REJECTED",
}
`;

exports[`ScansReducer should handle all defined error types: rejected types START_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "START_SCAN_REJECTED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types CANCEL_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "CANCEL_SCAN_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types DELETE_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "DELETE_SCAN_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types GET_REPORTS_DOWNLOAD 1`] = `
Object {
  "result": Object {
    "action": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_REPORTS_DOWNLOAD_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types GET_SCAN_CONNECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_CONNECTION_RESULTS_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types GET_SCAN_INSPECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_INSPECTION_RESULTS_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types GET_SCAN_JOB 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_JOB_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types GET_SCAN_JOBS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_JOBS_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types GET_SCANS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
  },
  "type": "GET_SCANS_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types PAUSE_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "PAUSE_SCAN_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types RESTART_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "RESTART_SCAN_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined fulfilled types: fulfilled types START_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "data": Object {
        "test": "success",
      },
      "date": undefined,
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": true,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": false,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "START_SCAN_FULFILLED",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types CANCEL_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "CANCEL_SCAN_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types DELETE_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "DELETE_SCAN_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types GET_REPORTS_DOWNLOAD 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_REPORTS_DOWNLOAD_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types GET_SCAN_CONNECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_CONNECTION_RESULTS_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types GET_SCAN_INSPECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_INSPECTION_RESULTS_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types GET_SCAN_JOB 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_JOB_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types GET_SCAN_JOBS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "GET_SCAN_JOBS_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types GET_SCANS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
  },
  "type": "GET_SCANS_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types PAUSE_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "PAUSE_SCAN_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types RESTART_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "RESTART_SCAN_PENDING",
}
`;

exports[`ScansReducer should handle all defined pending types: pending types START_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {
      "error": false,
      "errorMessage": "",
      "errorStatus": undefined,
      "fulfilled": false,
      "metaData": undefined,
      "metaId": undefined,
      "metaQuery": undefined,
      "pending": true,
      "update": false,
    },
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "START_SCAN_PENDING",
}
`;

exports[`ScansReducer should handle specific defined types: defined type DESELECT_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {
      "undefined": null,
    },
    "update": 0,
    "view": Object {},
  },
  "type": "DESELECT_SCAN",
}
`;

exports[`ScansReducer should handle specific defined types: defined type EXPANDED_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {
      "undefined": undefined,
    },
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "EXPANDED_SCAN",
}
`;

exports[`ScansReducer should handle specific defined types: defined type MERGE_SCAN_DIALOG_HIDE 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "MERGE_SCAN_DIALOG_HIDE",
}
`;

exports[`ScansReducer should handle specific defined types: defined type MERGE_SCAN_DIALOG_SHOW 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": undefined,
      "scans": undefined,
      "show": true,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "MERGE_SCAN_DIALOG_SHOW",
}
`;

exports[`ScansReducer should handle specific defined types: defined type NOT_EXPANDED_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {
      "undefined": null,
    },
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "NOT_EXPANDED_SCAN",
}
`;

exports[`ScansReducer should handle specific defined types: defined type RESET_SCAN_ACTIONS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 0,
    "view": Object {},
  },
  "type": "RESET_SCAN_ACTIONS",
}
`;

exports[`ScansReducer should handle specific defined types: defined type SELECT_SCAN 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {
      "undefined": undefined,
    },
    "update": 0,
    "view": Object {},
  },
  "type": "SELECT_SCAN",
}
`;

exports[`ScansReducer should handle specific defined types: defined type UPDATE_SCANS 1`] = `
Object {
  "result": Object {
    "action": Object {},
    "connection": Object {},
    "deleted": Object {},
    "expanded": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "mergeDialog": Object {
      "details": false,
      "scans": Array [],
      "show": false,
    },
    "selected": Object {},
    "update": 1654041600000,
    "view": Object {},
  },
  "type": "UPDATE_SCANS",
}
`;
